tests/robustness/model/history.go

```diff
@@ -195,35 +158,30 @@ func (h *AppendableHistory) AppendTxn(cmp []clientv3.Cmp, clientOnSuccessOps, cl
 	for _, resp := range resp.Responses {
 		results = append(results, toEtcdOperationResult(resp))
 	}
-	h.appendSuccessful(porcupine.Operation{
+	h.appendSuccessful(request, start, end, txnResponse(results, resp.Succeeded, revision))
+}
+
+func (h *AppendableHistory) appendSuccessful(request EtcdRequest, start, end time.Duration, response MaybeEtcdResponse) {
+	op := porcupine.Operation{
 		ClientId: h.streamID,
 		Input:    request,
 		Call:     start.Nanoseconds(),
-		Output:   txnResponse(results, resp.Succeeded, revision),
+		Output:   response,
 		Return:   end.Nanoseconds(),
-	})
-}
-
-func (h *AppendableHistory) appendSuccessful(op porcupine.Operation) {
+	}
 	if op.Call >= op.Return {
 		panic(fmt.Sprintf("Invalid operation, call(%d) >= return(%d)", op.Call, op.Return))
 	}
-	if len(h.successful) > 0 {
-		prevSuccessful := h.successful[len(h.successful)-1]
-		if op.Call <= prevSuccessful.Call {
-			panic(fmt.Sprintf("Out of order append, new.call(%d) <= prev.call(%d)", op.Call, prevSuccessful.Call))
+	if len(h.operations) > 0 {
```

Comment:
this section can be extracted to share with appendFailed.

Reply:
Ups, during refactors I botched my git history and dropped some commits. Let me restore it.